Sources/StreamVideoSwiftUI/CallView/ScreenSharing/ScreenSharingView.swift

```diff
@@ -73,23 +73,47 @@ public struct ScreenSharingView<Factory: ViewFactory>: View {
             contentMode: .scaleAspectFit
         ) { view in
             if let track = screenSharing.participant.screenshareTrack {
-                log.debug("adding screensharing track to a view \(view)")
+                log.info("Found \(track.kind) track:\(track.trackId) for \(screenSharing.participant.name) and will add on \(type(of: self)):\(identifier))", subsystems: .webRTC)
                 view.add(track: track)
             }
         }
-        .frame(
-            width: viewModel.hideUIElements ? videoSize.width : nil,
-            height: viewModel.hideUIElements ? videoSize.height : nil
-        )
+        .clipShape(RoundedRectangle(cornerRadius: 16))
     }
 
     private var videoSize: CGSize {
         if viewModel.hideUIElements {
-            return .init(width: availableFrame.size.height, height: availableFrame.size.width)
+            return .init(
+                width: frame.width,
+                height: frame.height - participantsStripFrame.height - innerItemSpace
+            )
+        } else {
+            return .init(
+                width: frame.width,
+                height: frame.height - innerItemSpace
+            )
+        }
+    }
+
+    private var itemsVisibleOnScreen: CGFloat {
+        if UIDevice.current.userInterfaceIdiom == .pad {
+            return UIDevice.current.orientation == .portrait ? 3 : 4
         } else {
-            return availableFrame.size
+            return 2
         }
     }
+
+    private var participantsStripFrame: CGRect {
+        /// Each video tile has an aspect ratio of 3:4 with width as base. Given that each tile has the
+        /// half width of the screen, the calculation below applies the aspect ratio to the expected width.
+        let aspectRatio: CGFloat = UIDevice.current.userInterfaceIdiom == .pad 
+        ? 9 / 16
```

Comment:
Indentation can be improved here, the operator shouldn't be on the same level as the let declaration.